library/asn1write.c

```diff
@@ -82,8 +82,12 @@ int mbedtls_asn1_write_len( unsigned char **p, unsigned char *start, size_t len
         *--(*p) = 0x83;
         return( 4 );
     }
-
+/*
+* relevant only for 64 bit platforms. on 32 bit platforms, size_t should always be 32 bits
+*/
+#if defined(MBEDTLS_HAVE_INT64)
```

Comment:
The macro `MBEDTLS_HAVE_INT64` is defined in `mbedtls/bignum.h`, but this header file is not included in this source file.